### PR TITLE
Added configurable iterator support to `ICache`

### LINKED ISSUE #7041: ICache.iterator() fetchSize is not configurable
ICache's iterator method returns an Iterator which fetches the 100 number of entries at a time. This number is hard coded now, and should be configurable.

### PR DESCRIPTION
**Pull request by serkan-ozal**
Closes #7041

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cache/ICache.java [hunk 1]
```diff
@@ -973,4 +975,32 @@
      */
     boolean removePartitionLostListener(String id);
 
+    /**
+     * <p>
+     * Creates and returns a cluster wide iterator
+     * to iterator on all entries owned by this cache.
```
Comment (emrahkocaman) [inline hazelcast/src/main/java/com/hazelcast/cache/ICache.java]: `to iterate` ?

**Comment by emrahkocaman** [comment]
:+1:

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientCacheIteratorTest.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 import javax.cache.spi.CachingProvider;
 
 
-@RunWith(HazelcastParallelClassRunner.class)
+@RunWith(HazelcastTestRunner.class)
```
Comment (emrahkocaman) [inline hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientCacheIteratorTest.java]: `@RunParallel` annotation should be added to run parameterized tests in parallel.

File: hazelcast/src/main/java/com/hazelcast/cache/ICache.java (hunk 1 above)
Comment (serkan-ozal) [inline hazelcast/src/main/java/com/hazelcast/cache/ICache.java]: fixed

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientCacheIteratorTest.java (hunk 2 above)
Comment (serkan-ozal) [inline hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientCacheIteratorTest.java]: done

**Comment by emrahkocaman** [comment]
:+1:

**Pull request merged**